src/compute/src/render/reduce.rs

```diff
@@ -292,11 +292,14 @@ where
     G::Timestamp: Lattice,
 {
     // we must have more than one arrangement to collate
-    soft_assert_or_log!(
-        arrangements.len() > 1,
-        "Building a collation of {} arrangements, but expected more than one",
-        arrangements.len()
-    );
+    if arrangements.len() <= 1 {
+        warn!("Building a collation of {} arrangements, but expected more than one in dataflow {debug_name}",
+            arrangements.len());
+        soft_assert_or_log!(
+            false,
```

Comment:
I wonder whether we should repeat the condition here to include them in the assertion text, or not. Maybe it's enough to have all the details in the warning beforehand, since someone needs to check the code anyways to figure out why the values violate an invariant?

Reply:
I had a similar feeling that it would be redundant, and preferred to avoid repeating the condition to avoid bugs due to potentially inconsistent changes. Happy to hear any suggestions if folks have a different opinion!

Reply:
I think this is probably ok but if we find ourselves using this pattern a lot, I'd prefer a macro that takes both the "verbose" and "Sentry" versions of the message and generates this, which would avoid the concerns around condition duplication.